AK/LEB128.h

```diff
@@ -28,7 +28,18 @@ class [[gnu::packed]] LEB128 {
     requires(Unsigned<ValueType>)
     {
         // First byte is unrolled for speed
-        auto byte = TRY(stream.read_value<u8>());
+        u8 byte;
+        do {
+            auto result = stream.read_some({ &byte, 1 });
+            if (result.is_error() && result.error().is_errno() && (result.error().code() == EAGAIN || result.error().code() == EINTR))
+                continue;
+            if (result.is_error())
+                return result.release_error();
+            if (result.value().size() != 1)
+                return Error::from_string_literal("EOF reached before expected end");
+            break;
+        } while (true);
```

Comment:
🤔 would it be possible to specialize this case for `Stream::read_value<T>` where `sizeof(T) == 1`? And if not, consider adding a comment to this section so no one's going for the obvious change back to `read_value<u8>()` to save some lines of code.